utility/BitVector.hpp

```diff
@@ -183,6 +183,19 @@ class BitVector {
   }
 
   /**
+   * @brief Assign this BitVector's contents to the pointed-to memory.
+   *
+   * @warning caller is responsible for ensuring the Bitvector has the correct
+   *          ownership and size.
+   *
+   * @param ptr Pointer to data representing a BitVector with the same parameters
+   *            as this BitVector.
+   **/
+  inline void setMemory(void *ptr) {
+    this->data_array_ = static_cast<std::size_t*>(ptr);
```

Comment:
Yes, it could if the `BitVector` owned memory and then is reassigned via this method. 

This certainly is a dangerous method but you can see why I used it right? Instead of having to instantiate an entirely new BitVector I essentially want to use BitVector's methods over a window of memory.

Adding a `DCHECK(owned)` here should demonstrate to programmers the proper usage.